partner_readonly_security/tests/test_partner_readonly_security.py

```diff
@@ -0,0 +1,86 @@
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(
+            context=dict(
+                cls.env.context,
+                mail_create_nolog=True,
+                mail_create_nosubscribe=True,
+                mail_notrack=True,
+                no_reset_password=True,
+                tracking_disable=True,
+                test_partner_readonly_security=True,
+            )
+        )
+        cls.user_admin = new_test_user(
+            cls.env,
+            login="test_user_admin",
+            groups="""
+                base.group_user,base.group_partner_manager,
+                partner_readonly_security.group_partner_edition
+            """,
+        )
+        cls.user_readonly = new_test_user(
+            cls.env,
+            login="test_user_readonly",
+            groups="base.group_user,base.group_partner_manager",
+        )
+        cls.partner = cls.env["res.partner"].sudo().create({"name": "Test partner"})
+
+    @users("test_user_admin")
+    @mute_logger("odoo.models.unlink")
+    def test_partner_admin(self):
+        """Read, write, unlink and create allowed."""
+        partners = self.env["res.partner"].search([])
+        self.assertIn(self.partner, partners)
+        self.partner.with_user(self.env.user).write({"name": "new-name"})
+        self.partner.with_user(self.env.user).unlink()
+        new_partner = self.env["res.partner"].create({"name": "Test partner 2"})
+        self.assertTrue(new_partner.exists())
+
+    @users("test_user_readonly")
+    def test_partner_readonly(self):
+        """Read allowed. Write, unlink and create not allowed."""
+        partners = self.env["res.partner"].search([])
+        self.assertIn(self.partner, partners)
+        with self.assertRaises(AccessError):
+            self.partner.with_user(self.env.user).write({"name": "new-name"})
+        with self.assertRaises(AccessError):
+            self.partner.with_user(self.env.user).unlink()
+        with self.assertRaises(AccessError):
+            self.env["res.partner"].create({"name": "Test partner 2"})
+
+    @users("test_user_readonly")
+    def test_partner_readonly_no_raise(self):
+        """Read allowed. Write, unlink and create return False
+        without exception when raise_exception=False."""
+        partners = self.env["res.partner"].search([])
+        self.assertIn(self.partner, partners)
+
+        result = self.partner.with_user(self.env.user).check_access_rights(
+            "write", raise_exception=False
+        )
+        self.assertFalse(result)
+
+        result = self.partner.with_user(self.env.user).check_access_rights(
+            "unlink", raise_exception=False
+        )
+        self.assertFalse(result)
+
+        result = (
+            self.env["res.partner"]
+            .with_user(self.env.user)
+            .check_access_rights("create", raise_exception=False)
+        )
+        self.assertFalse(result)
```

Comment:
This test is not necessary because it is testing the same as the _test_partner_readonly_ test.

Reply:
Previously I had received an error in the tests but I am trying again